libs/barista-components/filter-field/src/filter-field-config.ts

```diff
@@ -0,0 +1,32 @@
+/**
+ * @license
+ * Copyright 2020 Dynatrace LLC
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import { InjectionToken } from '@angular/core';
+import { createTagDataForFilterValues } from './filter-field-util';
+
+/** User defined configurations that can override some settings of the filter-field component */
+export interface DtFilterValuesConfig {
+  tagValuesParser: Function;
+}
+
+/** Injection token for the external configuration of the filter-field component */
+export const DT_FILTER_VALUES_CONFIG = new InjectionToken<DtFilterValuesConfig>(
```

Comment:
> We should also consider that the function may be passed into the filter field via an input.

This has been discussed previously and I would rather not add that one via an input. 
It does not need to change at runtime and could potentially be really hard to handle...

Reply:
> This has been discussed previously and I would rather not add that one via an input.
> It does not need to change at runtime and could potentially be really hard to handle...

The issue for me is, that without an additional input it is way harder to provide a parser function for just one filter-field without creating side effects for other filter-fields in child components.

e.g.

Root (defines parser function for all filter fields via DI
  - component (has filter field, uses root parser function because of DI)
    - child component (has filter field, needs different parser function which is set as a provider on this level)
      - child child component (has filter field, needs root parser function but inherits the function of its parent component even)

With an additional input the child component can set the special parser function via input and all other components (even child child component) inherit the correct root parser function.

Reply:
From what I see in the Barista documentation, there is a BaLiveExample component that hides by default the component decorator configuration, which is the place where the injection token would be provided. I don't see a way to configure the BaLiveExample component so that this configuration can be shown, so I guess that in order for the usage to be shown in the documentation, the component would have to be modified. If we go with the input approach however, this modification would not be needed.

Reply:
I'm still not happy about the input approach. This setting to me screams `configuration`, which I don't think should be changed at runtime. I understand the scoping problem of the DI, that you have outlined @thomaspink.

But if we have an input, and the input function changes, we would need to run though all active tag elements again and rerender them. This could maybe lead to some sort of flashing while the input is being evaluated. 

I'd still opt for the InjectionProvider... but if you think the input approach is feasible as well, I'm fine with that.

Reply:
Setting the function via DI should definitely be implemented. Not sure if i made it clear enough, the input should only be an extra option for these specific cases. So it is working the same way, for example, as the select uses the `ErrorStateMatcher` which can either be provided via DI or for special cases an Input.

I do not see the issue @tomheller has mentioned that the input function changes at runtime which could lead to rerendering the tags: 1. There is probably no use case for changing it at runtime. The function will be set on init in i guess all of the cases. Maybe you can help me with a use case where the runtime change is needed. 2. Even if the function changes at runtime, i do not see a problem. Input changes that causes a rerender in the case for 90% of the components. Also because the evaluation is done before rendering, i do not see why it would cause flashing. Maybe you can help me out with your thoughts there.

Regarding the effort, it should be kinda easy to add the input and override the value provided by the DI if an input is set. But I can agree to a compromise, as it handles only a specific use case, that we stick to the DI-only approach right now and add the input later if needed. :)

Reply:
> Not sure if i made it clear enough

Sorry, I guess I misunderstood and thought the input would be considered a replacement for the Di provided funciton. 

> 1. There is probably no use case for changing it at runtime.

There probably isn't a use case for it, but if it is there it should also work as expected, i.e. filters are set programmatically right on init and may be set before the function is set. But you are right, if the input changes is will cause re-rendering. I just wanted to make sure that it is not only re-renderd, but also the already committed tags run through the new function again. But as you said, this is probably not a very common use case, so I'm fine with the additional input to match the pattern of the `ErrorStateMatcher` 
👍

Reply:
Good points, both of you. I added an input option that takes priority over the injected configuration if it exists, let me know if this is what you had in mind.